### ISSUE TITLE
Re-subscription hangs forever

### ISSUE THREAD

**Issue by rocc0** [centrifugal/centrifuge-go#22]
Hi, I've found another bug, it happens when all environment goes down and than centrifugo  starts before signer endpoints. Client connects once and not subscribes, than hangs forever.
Steps:
1) turn off cent and endpoints
2) turn on cent, wait until client connects to the server
3) turn on endpoints(you can turn off/on only subscriptions signer)
Client yields `Connected with id: b7097fff-daf9-4f9a-adcd-eb8486912fb7,` and that's all.
It caused by:
```
	token, err := s.centrifuge.privateSign(s.channel)
	if err != nil {
		return err
	}

```

Because the status is not changed from `SUBSCRIBING` to `SUBERROR` or `UNSUBSCRIBER`  client will loop through this code forever:
```
	s.mu.Lock()
	if s.status == SUBSCRIBED || s.status == SUBSCRIBING {
		s.mu.Unlock()
		return nil
	}
```
I've fixed it for myself with: 
```
	token, err := s.centrifuge.privateSign(s.channel)
	if err != nil {
		s.mu.Lock()
		s.status = SUBERROR
		s.mu.Unlock()
		return err
	}
```
But i think that my fix is not convenient,  that's why I didn't create a PR. Also i've added pause on error to my subscription method, to prevent cent server from connection/disconnection flood that this fix causes.

**Comment by FZambia** [comment]
Hi, thanks a lot! Will try to reproduce and fix soon.

**Comment by FZambia** [comment]
@rocc0 please take a look at referenced pull request. When private subscription fails we now properly disconnect from Centrifugo and try to reconnect with backoff interval until we successfully connected and successfully resubscribed. We can discuss desired behaviour in that pull request.

To reproduce problem I used this code:

```
var now = time.Now().Unix()

func (h *eventHandler) OnPrivateSub(c *centrifuge.Client, e centrifuge.PrivateSubEvent) (string, error) {
	if time.Now().Unix()-now > 10 && time.Now().Unix()-now < 30 {
		return "", errors.New("boom")
	}
	token := subscribeToken(e.Channel, e.ClientID, time.Now().Unix()+10)
	return token, nil
}
```

I.e. temporary returning error from private sub handler after 10 secs from client start.

**Comment by FZambia** [comment]
Released with v0.2.2 - @rocc0 thanks a lot for your help!
